nifi-nar-bundles/nifi-aws-bundle/nifi-aws-abstract-processors/src/main/java/org/apache/nifi/processors/aws/v2/AwsClientCache.java

```diff
@@ -0,0 +1,41 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.nifi.processors.aws.v2;
+
+import com.github.benmanes.caffeine.cache.Cache;
+import com.github.benmanes.caffeine.cache.Caffeine;
+import org.apache.nifi.processor.ProcessContext;
+import software.amazon.awssdk.core.SdkClient;
+
+public class AwsClientCache<T extends SdkClient> {
+
+    private static final int MAXIMUM_CACHE_SIZE = 10;
```

Comment:
Any particular reason for the cache size of 10? It would be worth adding a comment of some kind.

Reply:
This was copied from the v1 version of the AwsClientCache, but looking at it now I don't see a strong reason to have a cache size limit.  Since the cache key boils down to the AWS region, multiple clients would be cached if a subclass of AbstractAwsProcessor parsed the region from the ProcessContext (as in the case of S3).  However, since there are currently only 38 regions, and likely not to be more than a few dozen regions in the future, it doesn't seem like a big risk to remove the limit.